lib/router.js

```diff
@@ -95,10 +95,12 @@ export default class Router {
     console.error(`No route found matching #${fragment}`);
   }
 
-  replaceHash(fragment) {
+  replaceHash(fragment, {historyMethod=null}={}) {
+    historyMethod = historyMethod || this.historyMethod;
     fragment = stripHash(fragment);
-    if (fragment !== stripHash(this.window.location.hash)) {
-      this.window.history[this.historyMethod](null, null, `#${fragment}`);
+    const currHash = stripHash(this.window.location.hash);
+    if (decodeURIComponent(fragment) !== decodeURIComponent(currHash)) {
```

Comment:
may want to add a comment here why we can't just compare fragment and currHash. Why is decodeURIComponent important here.

Reply:
added to a168ae4

Reply:
Do we need to uri-decode anywhere else? like https://github.com/mixpanel/panel/blob/549ddd9db965485016e7ef3db76038b60a556681/lib/router.js#L61-L62